File: mifosng-android/src/main/java/com/mifos/api/model/BulkRepaymentTransactions.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.api.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.Gson;

public class BulkRepaymentTransactions implements Parcelable {
    public int loanId;
    public double transactionAmount;

    //Optional fields

    private String accountNumber;
    private String bankNumber;
    private String checkNumber;
    private Integer paymentTypeId;
    private String receiptNumber;
    private String routingCode;

    public BulkRepaymentTransactions() {
    }

    public BulkRepaymentTransactions(int loanId, double transactionAmount) {
        this.loanId = loanId;
        this.transactionAmount = transactionAmount;
    }

    @Override
    public String toString() {
        return new Gson().toJson(this);
    }

    public int getLoanId() {
        return loanId;
    }

    public void setLoanId(int loanId) {
        this.loanId = loanId;
    }

    public double getTransactionAmount() {
        return transactionAmount;
    }

    public void setTransactionAmount(double transactionAmount) {
        this.transactionAmount = transactionAmount;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(String accountNumber) {
        this.accountNumber = accountNumber;
    }

    public String getBankNumber() {
        return bankNumber;
    }

    public void setBankNumber(String bankNumber) {
        this.bankNumber = bankNumber;
    }

    public String getCheckNumber() {
        return checkNumber;
    }

    public void setCheckNumber(String checkNumber) {
        this.checkNumber = checkNumber;
    }

    public Integer getPaymentTypeId() {
        return paymentTypeId;
    }

    public void setPaymentTypeId(Integer paymentTypeId) {
        this.paymentTypeId = paymentTypeId;
    }

    public String getReceiptNumber() {
        return receiptNumber;
    }

    public void setReceiptNumber(String receiptNumber) {
        this.receiptNumber = receiptNumber;
    }

    public String getRoutingCode() {
        return routingCode;
    }

    public void setRoutingCode(String routingCode) {
        this.routingCode = routingCode;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.loanId);
        dest.writeDouble(this.transactionAmount);
        dest.writeString(this.accountNumber);
        dest.writeString(this.bankNumber);
        dest.writeString(this.checkNumber);
        dest.writeValue(this.paymentTypeId);
        dest.writeString(this.receiptNumber);
        dest.writeString(this.routingCode);
    }

    protected BulkRepaymentTransactions(Parcel in) {
        this.loanId = in.readInt();
        this.transactionAmount = in.readDouble();
        this.accountNumber = in.readString();
        this.bankNumber = in.readString();
        this.checkNumber = in.readString();
        this.paymentTypeId = (Integer) in.readValue(Integer.class.getClassLoader());
        this.receiptNumber = in.readString();
        this.routingCode = in.readString();
    }

    public static final Parcelable.Creator<BulkRepaymentTransactions> CREATOR =
            new Parcelable.Creator<BulkRepaymentTransactions>() {
        @Override
        public BulkRepaymentTransactions createFromParcel(Parcel source) {
            return new BulkRepaymentTransactions(source);
        }

        @Override
        public BulkRepaymentTransactions[] newArray(int size) {
            return new BulkRepaymentTransactions[size];
        }
    };
}

File: mifosng-android/src/main/java/com/mifos/api/model/BulkRepaymentTransactions.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.api.model

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.Gson

data class BulkRepaymentTransactions(
    var loanId: Int = 0,
    var transactionAmount: Double= 0.0,
    // Optional fields
    var accountNumber: String? = null,
    var bankNumber: String? = null,
    var checkNumber: String? = null,
    var paymentTypeId: Int? = null,
    var receiptNumber: String? = null,
    var routingCode: String? = null
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readDouble(),
        parcel.readString(),
        parcel.readString(),
        parcel.readString(),
        parcel.readValue(Int::class.java.classLoader) as? Int,
        parcel.readString(),
        parcel.readString()
    )

    override fun toString(): String {
        return Gson().toJson(this)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(loanId)
        dest.writeDouble(transactionAmount)
        dest.writeString(accountNumber)
        dest.writeString(bankNumber)
        dest.writeString(checkNumber)
        dest.writeValue(paymentTypeId)
        dest.writeString(receiptNumber)
        dest.writeString(routingCode)
    }

    companion object CREATOR : Parcelable.Creator<BulkRepaymentTransactions> {
        override fun createFromParcel(parcel: Parcel): BulkRepaymentTransactions {
            return BulkRepaymentTransactions(parcel)
        }

        override fun newArray(size: Int): Array<BulkRepaymentTransactions?> {
            return arrayOfNulls(size)
        }
    }
}